Crash on Comparing Evian and other products

1. Scan barcode
2. Scan https://barcode.tec-it.com/en/EAN13?data=5997523313111 
3. Slide up and click compare
4. Scan https://barcode.tec-it.com/en/EAN13?data=3068320110783
5. BOOM

``` 
2019-03-07 16:47:40.991 21326-21326/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.openfoodfacts.scanner, PID: 21326
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.openfoodfacts.scanner/openfoodfacts.github.scrachx.openfood.views.ContinuousScanActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'openfoodfacts.github.scrachx.openfood.models.Product openfoodfacts.github.scrachx.openfood.models.State.getProduct()' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2951)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3086)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'openfoodfacts.github.scrachx.openfood.models.Product openfoodfacts.github.scrachx.openfood.models.State.getProduct()' on a null object reference
        at openfoodfacts.github.scrachx.openfood.views.product.ingredients.IngredientsProductFragment.a(Unknown Source:79)
        at android.support.v4.app.p.a(Unknown Source:342)
        at android.support.v4.app.p.e(Unknown Source:38)
        at android.support.v4.app.p.a(Unknown Source:46)
        at android.support.v4.app.p.f(Unknown Source:4)
        at android.support.v4.app.p.n(Unknown Source:6)
        at android.support.v4.app.m.e(Unknown Source:4)
        at android.support.v4.app.k.onCreate(Unknown Source:122)
        at android.support.v7.app.e.onCreate(Unknown Source:42)
        at openfoodfacts.github.scrachx.openfood.views.ContinuousScanActivity.onCreate(Unknown Source:7)
        at android.app.Activity.performCreate(Activity.java:7144)
        at android.app.Activity.performCreate(Activity.java:7135)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2931)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3086) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6718) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
```

``` 
2019-03-07 20:54:14.535 8584-8584/openfoodfacts.github.scrachx.openfood E/AndroidRuntime: FATAL EXCEPTION: main
    Process: openfoodfacts.github.scrachx.openfood, PID: 8584
    java.lang.RuntimeException: Unable to start activity ComponentInfo{openfoodfacts.github.scrachx.openfood/openfoodfacts.github.scrachx.openfood.views.ContinuousScanActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'openfoodfacts.github.scrachx.openfood.models.Product openfoodfacts.github.scrachx.openfood.models.State.getProduct()' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2951)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3086)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'openfoodfacts.github.scrachx.openfood.models.Product openfoodfacts.github.scrachx.openfood.models.State.getProduct()' on a null object reference
        at openfoodfacts.github.scrachx.openfood.views.product.ingredients.IngredientsProductFragment.onAttach(IngredientsProductFragment.java:186)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1372)
        at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1759)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1827)
        at android.support.v4.app.FragmentManagerImpl.dispatchStateChange(FragmentManager.java:3244)
        at android.support.v4.app.FragmentManagerImpl.dispatchCreate(FragmentManager.java:3194)
        at android.support.v4.app.FragmentController.dispatchCreate(FragmentController.java:184)
        at android.support.v4.app.FragmentActivity.onCreate(FragmentActivity.java:355)
        at android.support.v7.app.AppCompatActivity.onCreate(AppCompatActivity.java:84)
        at openfoodfacts.github.scrachx.openfood.views.ContinuousScanActivity.onCreate(ContinuousScanActivity.java:561)
        at android.app.Activity.performCreate(Activity.java:7144)
        at android.app.Activity.performCreate(Activity.java:7135)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2931)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3086) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6718) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ProductActivity.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ProductActivity.java
@@ -22,7 +22,6 @@ import android.support.design.widget.CoordinatorLayout;
 import android.support.design.widget.FloatingActionButton;
 import android.support.design.widget.TabLayout;
 import android.support.v4.app.ActivityCompat;
-import android.support.v4.app.FragmentActivity;
 import android.support.v4.content.ContextCompat;
 import android.support.v4.view.ViewPager;
 import android.support.v7.widget.LinearLayoutManager;
@@ -43,6 +42,7 @@ import android.widget.Toast;
 import com.afollestad.materialdialogs.MaterialDialog;
 
 import java.util.List;
+
 import butterknife.BindView;
 import butterknife.OnClick;
 import openfoodfacts.github.scrachx.openfood.BuildConfig;
@@ -269,14 +269,22 @@ public class ProductActivity extends BaseActivity implements OnRefreshListener {
 
 	private void setupViewPager( ViewPager viewPager )
 	{
+        adapterResult = new ProductFragmentPagerAdapter(getSupportFragmentManager());
 		adapterResult = setupViewPager(viewPager, adapterResult, mState, this);
     }
 
+    /**
+     * CAREFUL ! YOU MUST INSTANTIATE YOUR OWN ADAPTERRESULT BEFORE CALLING THIS METHOD
+     * @param viewPager
+     * @param adapterResult
+     * @param mState
+     * @param activity
+     * @return
+     */
     public static ProductFragmentPagerAdapter setupViewPager (ViewPager viewPager, ProductFragmentPagerAdapter adapterResult, State mState, Activity activity) {
         String[] menuTitles = activity.getResources().getStringArray( R.array.nav_drawer_items_product );
         String[] newMenuTitles=activity.getResources().getStringArray(R.array.nav_drawer_new_items_product);
 
-        adapterResult = new ProductFragmentPagerAdapter( ((FragmentActivity) activity).getSupportFragmentManager() );
         adapterResult.addFragment( new SummaryProductFragment(), menuTitles[0] );
         SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences( activity );
         if( BuildConfig.FLAVOR.equals( "off" ) || BuildConfig.FLAVOR.equals( "obf" ) || BuildConfig.FLAVOR.equals( "opff" ) )

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ProductFragment.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ProductFragment.java
@@ -206,6 +206,7 @@ public class ProductFragment extends Fragment implements OnRefreshListener {
     }
 
     private void setupViewPager(ViewPager viewPager) {
+        adapterResult = new ProductFragmentPagerAdapter(getChildFragmentManager());
         adapterResult = ProductActivity.setupViewPager(viewPager, adapterResult, mState, getActivity());
     }
 

